code/modules/clothing/suits/armor.dm

```diff
@@ -349,6 +349,15 @@
 	heat_protection = UPPER_TORSO | LOWER_TORSO | LEGS | FEET | ARMS | HANDS | HEAD
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
 
+/obj/item/clothing/suit/armor/reactive/fire/equipped(mob/user, slot)
+	..()
+	if(slot == slot_wear_suit)
+		ADD_TRAIT(user, TRAIT_RESISTHEAT, "incendiary_armor")
+
+/obj/item/clothing/suit/armor/reactive/fire/dropped(mob/user, silent)
+	..()
+	REMOVE_TRAIT(user, TRAIT_RESISTHEAT, "incendiary_armor")
+
```

Comment:
what if you pick up incendiary armour and drop it, while you have a set equipped. it will break it.
make the trait source, UID()

Reply:
done